src/Microsoft.AspNet.Mvc.Core/ActionResults/ViewExecutor.cs

```diff
@@ -23,22 +28,52 @@ public static class ViewExecutor
         /// <param name="actionContext">The <see cref="ActionContext"/> for the current executing action.</param>
         /// <param name="viewData">The <see cref="ViewDataDictionary"/> for the view being rendered.</param>
         /// <param name="tempData">The <see cref="ITempDataDictionary"/> for the view being rendered.</param>
-        /// <returns>A <see cref="Task"/> that represents the asychronous rendering.</returns>
+        /// <returns>A <see cref="Task"/> that represents the asynchronous rendering.</returns>
         public static async Task ExecuteAsync([NotNull] IView view,
                                               [NotNull] ActionContext actionContext,
                                               [NotNull] ViewDataDictionary viewData,
                                               [NotNull] ITempDataDictionary tempData,
-                                              string contentType)
+                                              MediaTypeHeaderValue contentType)
         {
-            if (string.IsNullOrEmpty(contentType))
+            var response = actionContext.HttpContext.Response;
+
+            // Encoding property on MediaTypeHeaderValue does not return the exact encoding instance that
+            // is set, so any settings(for example: BOM) on it will be lost when retrieving the value.
+            // In the scenario where the user does not supply encoding, we want to use UTF8 without BOM and
+            // for this reason do not rely on Encoding property.
+            MediaTypeHeaderValue contentTypeHeader;
```

Comment:
Could we verify what the behavior was in Mvc 5 - what happened when the `ContentEncoding` was set to `Encoding.Utf8` - did we write the BOM?